kernel/src/kernel.rs

```diff
@@ -877,15 +876,17 @@ impl Kernel {
                             subscribe_num: subdriver_number,
                         };
 
+                        // TODO: when the compiler supports capability types bring this back
+                        // as a NonNull type. https://github.com/tock/tock/issues/4134.
```

Comment:
```suggestion
                        // TODO: when the compiler supports capability types bring this back
                        // as a NonNull type. https://github.com/tock/tock/issues/4134.
                        // For now, checking for the null case is handled internally in each `map_or` call.
```

@LawrenceEsswood right?

Reply:
@alevy yes

@lschuermann 

Previously, we had a `NonNull` type (that had a niche) here, and could wrap that in `Option` to fill the niche and handle the `Null` case.

`CapabilityPtr` is filling the gap left by `* const()`, which does not have the niche and allows NULL internally. Having a CHERI capability type with a niche is (maybe?) predicated on having better compiler support. I think Option<NonNull<()>> is preferable here, and it should go back to it just as soon as we can express "non null capability".

Reply:
Okay, this makes sense. I think it'd help a great deal if you just copied that exact text into the comment. :smile: